### PR TITLE
Filter product properties after formatters

### PR DESCRIPTION
**Pull request by ghost**
There is currently no way to run a formatter function without including all properties required by it. Which can make objects larger than they need to be.

### CODE REVIEW

**Review by adamsoderstrom — COMMENTED** [review]
Hey, @janmysiak!

As mentioned in person, i had a hard time understanding what the problem was and what this solves.

Fill me in if i've got it wrong in the example below.

---

> Let's say you have a `product.relatedProducts` property.
>
> The `product.relatedProducts` property is an array of related products.
>
> Some of the entries in the `product.relatedProducts` property is of type `Variant`.
> 
> You want to expose the entries of `product.relatedProducts`, which are of the type `Variant` via a new property called `product.relatedVariantProducts`.
>
> You also want to remove the `product.relatedProducts` property.

If i've got it right: this isn't possible without your changes here, right?

**Comment by maeertin** [comment]
Could you have a look at this as well @alexanderflink? As I think you might have initially written the `formatProduct` function. Thanks

File: src/api/centra/utils/formatProduct.js [hunk 1]
```diff
@@ -25,7 +24,8 @@ export default function formatProduct(inProduct, inFormatters, filterProperties)
     product[optionKey] = typeof optionValue === 'function' ? optionValue(product) : optionValue
   })
 
-  return product
+  // filter out properties using filterProperties
+  return filterProperties ? filterProduct(product, filterProperties) : product
```
Comment (alexanderflink) [inline src/api/centra/utils/formatProduct.js:28]: Looks like this now requires the user to define their formatters twice, once in the formatters object and once in the filter array. Let's think about how we can solve this use case in a better way.

**Comment by maeertin** [comment]
Closing PR as "cia" is now purely a documentation app.

**Pull request closed without merging**